src/common/com/intellij/plugins/haxe/ide/index/HaxeComponentFileNameIndex.java

```diff
@@ -99,17 +101,26 @@ public static boolean getFileNames(@NotNull String qName,
     @NotNull
     public Map<String, Void> map(final FileContent inputData) {
       final PsiFile psiFile = inputData.getPsiFile();
+      final String moduleName = FileUtil.getNameWithoutExtension(inputData.getFileName());
       final List<HaxeClass> classes = HaxeResolveUtil.findComponentDeclarations(psiFile);
       if (classes.isEmpty()) {
         return Collections.emptyMap();
```

Comment:
> Main.hx which doesn't contains Main class, then we still relate that file to non-existant class, from which child classes can be accessed, right?

That's right

> Then everything else, seems like already implemented (by using something like findClassByQName)

+

> If file doesn't contains classes, what else useful it can provide? I mean typedef is considered to be a HaxeClass too in this system, if I got it right

For example Haxe compiler provide completion for `import` statement:
![image](https://cloud.githubusercontent.com/assets/3038174/12830463/80d6355c-cba0-11e5-8de4-ebe0ee08fe17.png)
![image](https://cloud.githubusercontent.com/assets/3038174/12830572/055a075e-cba1-11e5-979a-1e3d9f1aa917.png)